views/helpers.go

```diff
@@ -55,32 +71,119 @@ func DBToTemplateType(dbUser *[]user.User) []UserStripped {
-//	tplUsers.UserID = dbUser.UserID
-//	tplUsers.Username = dbUser.Username
-//	tplUsers.Nickname = dbUser.Nickname
-//	tplUsers.Name = dbUser.Firstname + " " + dbUser.Lastname
-//	tplUsers.Email = dbUser.Email
-//	tplUsers.Avatar = dbUser.Avatar
-//	tplUsers.UseGravatar = dbUser.UseGravatar
-//	if dbUser.LastLogin.Valid {
-//		tplUsers.LastLogin = dbUser.LastLogin.Time.Format("2006-01-02 15:04:05")
-//	} else {
-//		tplUsers.LastLogin = "-"
-//	}
-//	return tplUsers
-//}
+func DBUserToDetailedUser(dbUser user.User, store *user.Store) user.DetailedUser {
+	var u user.DetailedUser
+	var err error
+	u.UserID = dbUser.UserID
+	u.Username = dbUser.Username
+	u.UniversityUsername = dbUser.UniversityUsername
+	u.LDAPUsername = dbUser.LDAPUsername
+	u.LoginType = dbUser.LoginType
+	u.Nickname = dbUser.Nickname
+	u.Firstname = dbUser.Firstname
+	u.Lastname = dbUser.Lastname
+	u.Email = dbUser.Email
+	u.LastLogin = null.NewString(dbUser.LastLogin.Time.Format("2006-01-02 15:04:05"), dbUser.LastLogin.Valid)
+	u.ResetPw = dbUser.ResetPw
+	u.Enabled = dbUser.Enabled
+	u.CreatedAt = null.StringFrom(dbUser.CreatedAt.Time.Format("2006-01-02 15:04:05"))
+	if dbUser.CreatedBy.Valid {
+		u.CreatedBy, err = store.GetUser(context.Background(), user.User{UserID: int(dbUser.CreatedBy.Int64)})
+		if err != nil {
+			log.Println(err)
+			u.CreatedBy = user.User{
+				UserID:    int(dbUser.CreatedBy.Int64),
+				Firstname: "",
+				Nickname:  "",
+				Lastname:  "",
+			}
+		}
+	} else {
+		u.CreatedBy = user.User{
+			UserID:    -1,
+			Firstname: "",
+			Nickname:  "",
+			Lastname:  "",
+		}
+	}
+	if dbUser.UpdatedAt.Valid {
+		u.UpdatedAt = null.StringFrom(dbUser.UpdatedAt.Time.Format("2006-01-02 15:04:05"))
+	} else {
+		u.UpdatedAt = null.NewString("", false)
+	}
+	if dbUser.UpdatedBy.Valid {
+		u.UpdatedBy, err = store.GetUser(context.Background(), user.User{UserID: int(dbUser.UpdatedBy.Int64)})
+		if err != nil {
+			log.Println(err)
+			u.UpdatedBy = user.User{
+				UserID:    int(dbUser.UpdatedBy.Int64),
+				Firstname: "",
+				Nickname:  "",
+				Lastname:  "",
+			}
+		}
+	} else {
+		u.UpdatedBy = user.User{
+			UserID:    -1,
+			Firstname: "",
+			Nickname:  "",
+			Lastname:  "",
+		}
+	}
+	if dbUser.DeletedAt.Valid {
+		u.DeletedAt = null.StringFrom(dbUser.DeletedAt.Time.Format("2006-01-02 15:04:05"))
+	} else {
+		u.DeletedAt = null.NewString("", false)
+	}
+	if dbUser.DeletedBy.Valid {
+		u.DeletedBy, err = store.GetUser(context.Background(), user.User{UserID: int(dbUser.DeletedBy.Int64)})
+		if err != nil {
+			log.Println(err)
+			u.DeletedBy = user.User{
+				UserID:    int(dbUser.DeletedBy.Int64),
+				Firstname: "",
+				Nickname:  "",
+				Lastname:  "",
+			}
+		}
+	} else {
+		u.DeletedBy = user.User{
+			UserID:    -1,
+			Firstname: "",
+			Nickname:  "",
+			Lastname:  "",
+		}
+	}
+	if dbUser.UseGravatar {
+		u.UseGravatar = true
+		hash := md5.Sum([]byte(strings.ToLower(strings.TrimSpace("liam.burnand@bswdi.co.uk"))))
```

Comment:
Hard-coded

Reply:
I think that was a mistake merge from sorting this PR out but was sorted in PR-10 and now